library/kotlin/io/envoyproxy/envoymobile/stats/TimerImpl.kt

```diff
@@ -7,15 +7,27 @@ import java.lang.ref.WeakReference
  * Envoy implementation of a `Timer` for time measurements e.g. distribution of durations.
  */
 internal class TimerImpl : Timer {
-  internal val envoyEngine: WeakReference<EnvoyEngine>
-  internal val elements: String
+  var envoyEngine: WeakReference<EnvoyEngine>
+  var series: String
+  var tags: Map<String, String>
 
-  internal constructor(engine: EnvoyEngine, elements: List<Element>) {
+  internal constructor(engine: EnvoyEngine, elements: List<Element>, tags: List<Tag>) {
     this.envoyEngine = WeakReference<EnvoyEngine>(engine)
-    this.elements = elements.joinToString(separator = ".") { it.value }
+    this.series = elements.joinToString(separator = ".") { it.value }
+    this.tags = convert(tags)
+  }
+
+  private constructor(engine: WeakReference<EnvoyEngine>, series: String, tags: List<Tag>) {
+    this.envoyEngine = engine
+    this.series = series
+    this.tags = convert(tags)
   }
 
   override fun completeWithDuration(durationMs: Int) {
```

Comment:
unrelated nit: I _really_ would like to see this method renamed to `recordDuration`; I think the current name is quite misleading.

Reply:
i created a [ticket](https://jira.lyft.net/browse/RES-2173) for this one